api/app/domain/analyzers/land_cover_change_analyzer.py

```diff
@@ -0,0 +1,159 @@
+        4: "Water",
+        5: "Snow/ice",
+        6: "Cropland",
+        7: "Built-up",
+        8: "Cultivated grasslands",
+    }
+
+    number_of_classes = len(land_cover_mapping)
+
+    def __init__(
+        self,
+        analysis_repository=None,
+        compute_engine=None,
+        dataset_repository=None,
+    ):
+        self.analysis_repository = analysis_repository  # LandCoverChangeRepository
+        self.compute_engine = compute_engine  # Dask Client, or not?
+        self.dataset_repository = dataset_repository  # AWS-S3 for zarrs, etc.
+        self.admin_results_uri = "s3://gfw-data-lake/umd_lcl_land_cover/v2/tabular/statistics/admin_land_cover_change.parquet"
+        self.land_cover_zarr_uri = "s3://gfw-data-lake/umd_lcl_land_cover/v2/raster/epsg-4326/zarr/umd_lcl_land_cover_2015-2024.zarr/"
+        self.pixel_area_zarr_uri = "s3://gfw-data-lake/umd_area_2013/v1.10/raster/epsg-4326/zarr/pixel_area.zarr/"
+
+    async def analyze(self, analysis: Analysis):
+        land_cover_change_analytics_in = LandCoverChangeAnalyticsIn(**analysis.metadata)
+        if land_cover_change_analytics_in.aoi.type == "admin":
+            gadm_ids = land_cover_change_analytics_in.aoi.ids
+            combined_results_df = self.analyze_admin_areas(gadm_ids)
+
+        else:
+            aois = land_cover_change_analytics_in.aoi.model_dump()
+            geojsons = await get_geojson(aois)
+            if aois["type"] != "feature_collection":
+                aoi_list = sorted(
+                    [{"type": aois["type"], "id": id} for id in aois["ids"]],
+                    key=lambda aoi: aoi["id"],
+                )
+            else:
+                aoi_list = aois["feature_collection"]["features"]
+                geojsons = [geojson["geometry"] for geojson in geojsons]
+
+            analysis_partial = partial(
+                self.analyze_area,
+                land_cover_zarr_uri=self.land_cover_zarr_uri,
+                pixel_area_zarr_uri=self.pixel_area_zarr_uri,
+            )
+            dd_df_futures = await self.compute_engine.gather(
+                self.compute_engine.map(analysis_partial, aoi_list, geojsons)
+            )
+            dfs = await self.compute_engine.gather(dd_df_futures)
+            combined_results_df = await self.compute_engine.compute(dd.concat(dfs))
+
+        combined_results_df = combined_results_df[
+            combined_results_df.land_cover_change_area__ha > 0
+        ]
+        analyzed_analysis = Analysis(
+            combined_results_df.to_dict(orient="list"),
+            analysis.metadata,
+            AnalysisStatus.saved,
+        )
+        await self.analysis_repository.store_analysis(
+            land_cover_change_analytics_in.thumbprint(), analyzed_analysis
+        )
+
+    def analyze_admin_areas(self, gadm_ids):
+        query = f"select * from '{self.admin_results_uri}' where aoi_id in {gadm_ids} and land_cover_class_start != land_cover_class_end"
```

Comment:
Is the idea here not to report on when land cover hasn't changed? I feel like that also would be interesting to know. E.g. amount of stable forest vs forest that got converted.

Reply:
I can clarify for requirements